PlottingScripts/Basic_Plot.py

```diff
@@ -0,0 +1,128 @@
+"""This is a basic script for plotting an analysis file."""
+from collections import OrderedDict
+
+import numpy as np
+import pandas as pd
+import pdb
+
+from bokeh.plotting import figure, show, output_file
+
+
+"""This script is just the first step in visualizing the data set. This file
+comprises of the method creating an html bokeh plot and a command at the end to
+implement the method
+"""
+
+
+def makeplot(Dataframe, Cols, minvalues):
+    """Basic method to plot sensitivity anlaysis.
+
+    This is the method to generate a bokeh plot followung the burtin example
+    template at the bokeh website. For clarification, parameters refer to an
+    output being measured (Tmax, C, rxn2, etc.) and stats refer to the 1st or
+    total order sensitivity index.
+
+    Inputs:
+    Dataframe - Name of csv file to be read into as a panda dataframe (String)
+                **Note file must be in same folder as script.**
+    Cols- Names of the columns to be plotted (Array of strings)
+    minvalues- Cutoff minimum for which parameters should be plotted (Array of
+                floats)
+    Outputs:
+    Generates an html of the plot. The axis is a log scale.
+    """
+    # Read in csv file as panda dataframe.
+    tdf = pd.read_csv(Dataframe, delimiter=' ', skipinitialspace=True,
+                      engine='python')
+    df = tdf
+    maxval = 0
+    # Remove rows which have values less than cutoff values
+    for i in range(0, minvalues.size):
+        df = df[df[Cols[i]] > minvalues[i]]
+        if maxval < max(df[Cols[i]]):
+            maxval = max(df[Cols[i]])
+    df = df.dropna()
+
+    # Create dictionary of bar colors, assume up to 3 stats are plotted
+    # per parameter.
+    colors = ["#0d3362", "#c64737", "black"]
+    stat_color = OrderedDict()
+    for i in range(0, Cols.size):
+        stat_color[i] = colors[i]
+    # Reset index of dataframe.
+    df = df.reset_index(drop=True)
+
+    # Sizing parameters
+    width = 800
+    height = 800
+    inner_radius = 90
+    outer_radius = 300 - 10
+
+    # Determine wedge size based off number of parameters
+    big_angle = 2.0 * np.pi / (len(df)+1)
+    # Determine division of wedges for plotting bars based on # stats plotted
```

Comment:
Ah, I see.  Looking at that example in particular it is clear that this is
a problem when the value of a sensitivity index is smaller than it's
confidence range and the value is close to zero.  This shouldn't be a
problem with any stacked bar type plots because differences that small
won't show, and even if they do then the error bars on the plot will
indicate what is going on.

On Wed, Feb 24, 2016 at 9:53 AM, Chris Fu <notifications@github.com> wrote:

> In PlottingScripts/Basic_Plot.py
> <https://github.com/houghb/HDSAviz/pull/2#discussion_r53977810>:
>
> > +    colors = ["#0d3362", "#c64737", "black"]
> > +    stat_color = OrderedDict()
> > +    for i in range(0, Cols.size):
> > +        stat_color[i] = colors[i]
> > +    # Reset index of dataframe.
> > +    df = df.reset_index(drop=True)
> > +
> > +    # Sizing parameters
> > +    width = 800
> > +    height = 800
> > +    inner_radius = 90
> > +    outer_radius = 300 - 10
> > +
> > +    # Determine wedge size based off number of parameters
> > +    big_angle = 2.0 * np.pi / (len(df)+1)
> > +    # Determine division of wedges for plotting bars based on # stats plotted
>
> I have seen instances where S1 is greater than ST (rxn107 in
> analysis_CO2.txt is one example). Ordering the the values shouldn't be too
> difficult because it should just require re-ordering the panda dataframe by
> a user specified column. Perhaps this would best be implemented as a
> separate function though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/houghb/HDSAviz/pull/2/files#r53977810>.
>